### ISSUE TITLE
Document non-snap installation instructions

### ISSUE THREAD

**Issue by jixunmoe** [certbot/certbot#8345]
## My operating system is (include version):

Ubuntu 20.04

## Certbot's behavior differed from what I expected because:

I expect the official installation guide to have instructions for me to install certbot without using snap.

When I [select `nginx` and `Ubuntu 20.04` from the dropdown](https://certbot.eff.org/lets-encrypt/ubuntufocal-nginx), I was greeted with the following message:

> ⚠️ Snap Support
> The Certbot snap supports the x86_64, ARMv7, and ARMv8 architectures. You can find instructions for installing Certbot without using snap by selecting your OS in the dropdown above.

<details>
<summary>Screenshot of the page</summary>
<img src="https://user-images.githubusercontent.com/5713045/95015061-cd7c0f80-0642-11eb-97f5-76cf8baecd5e.png">
</details>

However looking under the OS dropdown, _none_ of the options resembles "Ubuntu without snap" .

![image](https://user-images.githubusercontent.com/5713045/95014844-8a6d6c80-0641-11eb-9d1f-939e8d094952.png)

I think the instructions page should be more clear where to find those. Provide a link instead would help, for instance:

> ⚠️ Snap Support
> The Certbot snap supports the x86_64, ARMv7, and ARMv8 architectures. 
> You can find instructions for installing Certbot without using snap by [clicking here](https://example.com).

![image](https://user-images.githubusercontent.com/5713045/95015171-6ca10700-0643-11eb-9f0c-b6314a05ff59.png)

If the traditional package manager installation method is deprecated or unmaintained anymore, please indicate it somewhere.

**Comment by alexzorin** [comment]
Somewhat related: https://github.com/certbot/website/issues/636

**Comment by twig** [comment]
This would also be incredibly useful for WSL Ubuntu 20 as snap is not currently available.

**Comment by manualmanul** [comment]
Snaps aren't well supported within LXC containers, so this would be a welcome change.

**Comment by Aaron-ML** [comment]
Also a huge blocker to use snap in centos7/centos8 with selinux packages missing. 

Alternatives need to be documented.

**Comment by rustyx** [comment]
```
wget https://dl.eff.org/certbot-auto
sudo mv certbot-auto /usr/local/bin/certbot-auto
sudo chown root /usr/local/bin/certbot-auto
sudo chmod 0755 /usr/local/bin/certbot-auto
sudo certbot-auto
```

Source: https://certbot.eff.org/docs/install.html#certbot-auto

**Comment by Total-Ecl1ps3** [comment]
To add, it is really annoying that the non-snap guide _has existed_ before, but now has been removed completely.

Telling me to find instructions for non-snap installation by "selecting my OS" doesn't exactly make it any better, since I've already selected my OS.

If you look at the table below, i compared every option in the "OS" Dropdown:
If i were to try to find the Ubuntu install guide, but using apt instead of snap, i would have to know that Devuan distro's also use apt, which is just confusing.

Other than everything mentioned above, why did anyone even bother writing in all the separate options if they **all** use the same "snap" guide, while also adding "snapd" as a separate option in the OS list....

<table border="solid 1px black">
<thead>
    <tr>
        <th>Distro</th>
        <th>OS option</th>
        <th>Method</th>
    </tr>
</thead>
<tbody>
    <tr>
        <td>Snap</td><td>snapd</td><td rowspan="15">snap</td>
    </tr>
    <tr>
        <td rowspan="3">Debian</td><td>Debian 9 (stretch)</td>
    </tr>
    <tr>
        <td>Debian 10 (buster)</td>
    </tr>
    <tr>
        <td>Debian testing/unstable</td>
    </tr>
    <tr>
        <td rowspan="4">Ubuntu</td><td>Ubuntu 20.04</td>
    </tr>
    <tr>
        <td>Ubuntu 19.10</td>
    </tr>
    <tr>
        <td>Ubuntu 18.04 LTS (bionic)</td>
    </tr>
    <tr>
        <td>Ubuntu 16.04 (xenial)</td>
    </tr>
    <tr>
        <td>Arch</td><td>Arch Linux</td>
    </tr>
    <tr>
        <td>Fedora</td><td>Fedora</td>
    </tr>
    <tr>
        <td rowspan="2">openSUSE</td><td>openSUSE Tumbleweed</td>
    </tr>
    <tr>
        <td>openSUSE Leap 15.0</td>
    </tr>
    <tr>
        <td>Other</td><td>Other Linux</td>
    </tr>
    <tr>
        <td rowspan="4">RHEL/CentOS</td><td>CentOS/RHEL 7</td>
    </tr>
    <tr>
        <td>CentOS/RHEL 8</td>
    </tr>
    <tr>
        <td>CentOS 6</td><td rowspan="2">wget</td>
    </tr>
    <tr>
        <td>RHEL 6</td>
    </tr>
    <tr>
        <td>Gentoo</td><td>Gentoo</td><td>emerge</td>
    </tr>
    <tr>
        <td>FreeBSD</td><td>FreeBSD</td><td>pkg/ports</td>
    </tr>
    <tr>
        <td>OpenBSD</td><td>OpenBSD 6.0+</td><td>pkg_add</td>
    </tr>
    <tr>
        <td>macOS</td><td>macOS</td><td>Homebrew</td>
    </tr>
    <tr>
        <td rowspan="3">Devuan</td><td>Devuan ASCII 2.0</td><td rowspan="3">apt-get</td>
    </tr>
    <tr>
        <td>Devuan Beowulf 3.0</td>
    </tr>
    <tr>
        <td>Devuan testing/unstable</td>
    </tr>
    <tr>
        <td>Windows</td><td>Windows</td><td>exe</td>
    </tr>
</tbody>
</table>

**Comment by ygoe** [comment]
That certbot-auto script has only been tested with Ubuntu versions as old as 14.04 and 15.04 (noted in lines 314 to 315). This doesn't look too promising. I'm still trying to read that install script to find the actual installation instructions for a modern OS without snap. All I'm looking for is a way to get the certbot program and run it. No automatic updates that possibly try to install extras that would have been included 6 years ago.

From my past experience, I thought this would be simple on my new server. I scheduled 2 evenings to get everything up and running – including my management application that needs to be rewritten for a new environment.

Maybe certbot isn't that simple and I need something entirely different.

**Comment by progreccor** [comment]
> 
> Maybe certbot isn't that simple and I need something entirely different.
try to use acme.sh ;)

**Comment by ygoe** [comment]
Thanks, I've already found that. After further exploring my requirements, I found that a custom client based on a program library is better for me. I can integrate that into my custom management application that already sits on the domain and DNS control.

I think that the strong Certbot recommendation on the LetsEncrypt.org website is a bit misplaced. acme.sh might better be mentioned there, gived that it has no dependencies and should be very portable (not sure about Windows here).

**Comment by progreccor** [comment]
> I think that the strong Certbot recommendation on the LetsEncrypt.org website is a bit misplaced. acme.sh might better be mentioned there, given that it has no dependencies and should be very portable (not sure about Windows here).

Also see this link https://www.xf.is/2020/06/30/list-of-free-acme-ssl-providers/

**Comment by alyandon** [comment]
I just received a message today that certbot-auto is no longer supported on my system and the only suggested replacement is the snap package.  I'm not going to run snapd and give up basic abilities like being able to control when packages get updated, being able to see the complete update history of a package, etc.  Some clarity on how to install the non-snap version of the certbot client would be greatly appreciated.

**Comment by dsoriano** [comment]
I will not install this shit of snapd on my servers. If we don't have any other solution to continue to use certbot without snapd, I will give it up.

**Comment by progreccor** [comment]
> I will not install this shit of snapd on my servers. If we don't have any other solution to continue to use certbot without snapd, I will give it up.

simply use acme.sh 
https://github.com/acmesh-official/acme.sh

**Comment by bmw** [comment]
Other options for installing Certbot are documented at https://certbot.eff.org/docs/install.html.

certbot-auto is still listed there, but it currently only works on RHEL based systems, but that will change in our next planned release.

**Comment by ygoe** [comment]
Who is responsible for getting a current release into the Ubuntu packages? There's still 0.40 which seems a little outdated now. Of course it would be appreciated if Ubuntu (or any distro) came with a usable package on its own, so we wouldn't have to mess with strange custom installation methods. But I know of some vendors that start their installation instructions by uninstalling distro packages.

**Comment by bmw** [comment]
You might be interested in https://github.com/certbot/certbot/issues/8239.

**Comment by joeysundotcom** [comment]
apparently, opensuse 15.2 can get certbot by running
`sudo zypper in python3-certbot`
this is also missing in the documentation.

why the **** would i want to install a package manager, that needs to run a daemon and a socket, when there is one, that comes with the OS?!?

it's advanced douchebaggery, and you know it.
stop being dumb.

**Comment by DispatchCommit** [comment]
I'm glad you guys kept mentioning acme.sh, it's great! IMO, it's much easier to use than even certbot, all the arguments are shorter and easier to remember, and the instructions / documentation were written so well that I had things working on my first attempt.

I didn't want to use apt & snapd and was frustrated at how hard it has become to just install certbot compared to how things were when I first started using it. Little disappointed that there wasn't even an option to use apt OR snap. And it doesn't seem to me that there is a clear reason / explanation provided to devs as to why their only choice is snap and not apt anymore.

**Comment by bmw** [comment]
I think this issue can be closed. On https://certbot.eff.org/instructions, there are instructions on how to install using `pip` and the snap instructions start by linking to https://certbot.eff.org/docs/install.html where there are alternate installation methods documented such as Docker and OS packages.

**Comment by cazort** [comment]
I don't consider this resolved. The page you link to prompts for the user's distribution (pip is not a distribution) and then the resulting page in most cases does not show any alternative options beyond snap. Furthermore, the page for pip has a disclaimer that it only has "partial support".

At a bare minimum, this is somewhat unclear documentation that makes it hard for people looking for non-snap-based installation methods to find what they want.

It is also clear that the team running this project is pushing users in the direction of snap, in spite of nearly unanimous kickback from the userbase. I.e. there may be plenty of users _indifferent_ to using snap, but I don't see any actively supporting its use, and I see a lot of people who actively want to avoid it.

It is especially concerning to me that the main objections to snap are ones people raise about production servers, and this is the environment where it is most likely someone would want to use certbot.

This issue is driving me to want to discontinue my use of certbot and migrate towards other tools. However, given how LetsEncrypt recommends certbot as their #1 recommendation, this is also souring my feelings towards LetsEncrypt. It's like the teams running these things don't really have my interests or needs in mind.

**Comment by phil294** [comment]
@bmw could you please reopen this?

**Comment by harmin-parra** [comment]
I agree with the last comments.
This issue has not been solved and needs to be reopened.

**Comment by mumblingsages** [comment]
I agree as well.

**Comment by Radiergummi** [comment]
@bmw the "alternate installation methods" documented feel like unwarranted mockery. Currently, the page you linked lists:

 1. **Snap**: Snap as an alternative to Snap? Interesting. The whole page includes no less than six links to the ordinary Snap installation page. This just feels patronising, insulting to my experience as a systems administrator, and to my conscious choice of looking for alternative installation methods in the first place.
 2. **Docker**: Explicitly described as unable to configure my web server, practically ruling this option out. Additionally, the full text reads more as a warning of the method than instructions - see point 1.
 3. **Certbot-Auto**: No longer supported, ruling this out, too.
 4. **Pip**: Supported on "a best effort basis", eg. not at all.

In conclusion, there are no alternative installation methods. I'd just wish for the Certbot team to have the courage to plainly say "there are no alternatives, use Snap or migrate to another ACME client", and sparing everyone the time and effort of discovering this one GitHub issue at a time.

**Comment by DXXS** [comment]
Yep; snap requires fuse kernel module(s) which neither of my vps providers have been willing to add; therefore I ended up reading between the lines on the pip installation & eventually got a dns variant of 1.27 installed!

**Comment by donatj** [comment]
This should indeed be reopened. The whole reason I went with Debian rather than Ubuntu while setting up my new server was to not have to stay away from Snap - and now you're trying to force snap on me just to use certbot? That's silly.
